src/System.Private.ServiceModel/src/System/IdentityModel/Selectors/SecurityTokenProvider.cs

```diff
@@ -58,6 +69,8 @@ public virtual bool SupportsTokenCancellation
         }
 
         // protected methods
+        protected abstract SecurityToken GetTokenCore(TimeSpan timeout);
```

Comment:
I'm wondering if we should just omit this and utilize GetTokenCoreAsync exclusively through our .NET Core code. Maintaining two codepaths for effectively the same thing isn't ideal 

Furthermore, the things we're porting over right now seem not to use the timeout parameter, though I did see it in S.SM.IssuanceTokenProvider and S.SM.SecuritySessionSecurityTokenProvider for .NET Desktop (but then this seems like something we might want to rewrite using task async anyway)